### PR TITLE
feat(ilp): make ILP over HTTP transactional for single table

### PR DESCRIPTION
**Pull request by ideoma**
This change will make ILP over HTTP more transactional a single table message. This is achieved by 

- removing automatic commit after X lines per table threshold (this does not affect ILP over TCP)
- removing artificial data commit before a new column has to be added to the table

Now if you send any size of data to ILP over HTTP for a single table it will be all in or all out.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 24 / 24 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/LineHttpProcessorState.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/WalTableUpdateDetails.java|1|1|100.00%|

File: core/src/main/java/io/questdb/cairo/wal/WalWriter.java [hunk 1]
```diff
@@ -491,10 +493,13 @@ public void rollUncommittedToNewSegment() {
 
                 try {
                     final int timestampIndex = metadata.getTimestampIndex();
+
                     LOG.info().$("rolling uncommitted rows to new segment [wal=")
                             .$(path).$(Files.SEPARATOR).$(oldSegmentId)
+                            .$(", segTxn=").$(lastSegmentTxn)
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/wal/WalWriter.java:499]: nit: there is no reason not to align logging with variable names

File: core/src/main/java/io/questdb/cutlass/line/tcp/IlpWalAppender.java [hunk 2]
```diff
@@ -150,11 +146,13 @@ private void appendToWal0(
                         columnWriterIndex = metadata.getColumnIndexQuiet(columnNameUtf16);
                         if (columnWriterIndex < 0) {
                             securityContext.authorizeAlterTableAddColumn(writer.getTableToken());
-                            tud.commit(false);
                             try {
                                 writer.addColumn(columnNameUtf16, ld.getColumnType(ld.getColNameUtf8(), ent.getType()), securityContext);
                                 columnWriterIndex = metadata.getColumnIndexQuiet(columnNameUtf16);
                             } catch (CairoException e) {
+//                                if (e.isTableDropped()) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cutlass/line/tcp/IlpWalAppender.java:153]: dead code

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 25 / 25 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/LineHttpProcessorState.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/WalTableUpdateDetails.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 25 / 25 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/LineHttpProcessorState.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/WalTableUpdateDetails.java|1|1|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**